resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -70,6 +73,11 @@ private[spark] class Client(
 
   private val isClusterMode = sparkConf.get("spark.submit.deployMode", "client") == "cluster"
 
+  private val isClientUnmanagedAMEnabled = sparkConf.get(YARN_UNMANAGED_AM) && !isClusterMode
+  private var amServiceStarted = false
+  private var appMaster: ApplicationMaster = _
+  private var unManagedAMStagingDirPath: Path = _
```

Comment:
Seems better to just store this in a variable for all cases. It's recomputed from the conf in a few different places in this class.